### PR TITLE
Only include as many domains as we have in the sorted array

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

A very simple PR, see title. Fixes https://github.com/pi-hole/FTL/issues/1812

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: src/api/stats.c [hunk 1]
```diff
@@ -160,10 +150,23 @@ int api_stats_top_domains(struct ftl_conn *api)
...
+	// Allocate memory
+	int count = 10;
+	bool audit = false;
+	const int domains = counters->domains;
+	int *temparray = calloc(2*domains, sizeof(int*));
+	if(temparray == NULL)
+	{
+		log_err("Memory allocation failed in %s()", __FUNCTION__);
+		return 0;
+	}
```
Comment (DL6ER) [inline src/api/stats.c:168]: This code has been moved down to after the `if` so we don't even have to allocate the memory when exiting anyway due to privacy settings. I also moved the shared memory locking up so we can be sure `counter->domains` does not change while we are doing this here (race-collision prevention).

**Comment by PromoFaux** [comment]
@yubiuser can you confirm this fixes the issue you reported in #1812 ?

**Comment by yubiuser** [comment]
No, I can't, but only because I don't know how to trigger the issue. I only saw it once and don't know which conditions triggered it.

**Review by yubiuser — APPROVED** [review]
Code looks sane and list of blocked domains is returned as expected.

**Pull request merged**